docs/api.yaml

```diff
@@ -0,0 +1,1061 @@
+          required: true
+          schema:
+            type: string
+      responses:
+        200:
+          description: "Teacher deleted"
+        400:
+          $ref: "#/components/responses/400"
+        403:
+          $ref: "#/components/responses/403"
+        404:
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+components:
+  #securitySchemes:
+  # bearerAuth:
+  # type: http
+  # scheme: bearer
+  responses:
+    400:
+      description: Bad request
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/BadRequestException"
+    403:
+      description: Forbidden
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/NotAuthorizedException"
+    404:
+      description: Not found
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/ResourceNotFoundException"
+    429:
+      description: Too many requests to the API
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/TooManyRequestsException"
+    500:
+      description: Internal server error
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/InternalServerException"
+
+  schemas:
+
+    ######################################################
+    ####################### GRADE ########################
+    ######################################################
+
+    Grade:
+      type: object
+      properties:
+        id:
+          type: string
+        value:
+          type: integer
+        enrolled_course_id:
+          type: string
+        enrolled_course_name:
+          type: string
+
+    ######################################################
+    ####################### STUDENT ######################
+    ######################################################
+
+    Student:
+      type: object
+      properties:
+        id:
+          type: string
+        first_name:
+          type: string
+        last_name:
+          type: string
+        sex:
+          type: string
+          enum:
+            - M
+            - F
+        birth_date:
+          type: "string"
+          format: "date"
+        address:
+          type : string
+        phone_number:
+          type: string
+        email:
+          type: string
+        entrance_date:
+          type: "string"
+          format: "date"
+        status:
+          type: string
+          description: "Defaults to ENABLED"
+          enum:
+            - ENABLED
+            - DISABLED
+        group:
+          $ref: '#/components/schemas/Group'
+        enrolled_courses:
+          type: array
+          items:
+            $ref: '#/components/schemas/EnrolledCourse'
+        grades:
```

Comment:
NO. grade is just an attribute of  an enrolled_course. Just add grade to the objects of the enrolled_courses list...

Reply:
Can a course own many grades ? 
For example : 
When a student fails, is it necessary to have a history of the grades by date, so we create an object of type date with its value and its date and choose the most recent.

Or

We just update the grade value so the grade isn't an object but just an attribute of type integer ?

Reply:
excellent point. we indeed want to keep trace of the number of attempts before succeeding. Hence, instead ot the enrolled_course.grade, put enrolled_course.exams